src/App.tsx

```diff
@@ -1,27 +1,83 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
+import Button from 'react-bootstrap/Button';
+import ButtonGroup from 'react-bootstrap/ButtonGroup';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Loader } from './Components/Loader';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goodsList, setGoodsList] = useState<Good[]>([]);
+  const [waiter, setWaiter] = useState(false);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const handleGetAll = () => {
+    setWaiter(true);
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+    setTimeout(async () => {
+      setWaiter(false);
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+      return setGoodsList(await getAll());
+    }, 500);
+  };
 
-    <GoodsList goods={[]} />
-  </div>
-);
+  const handleGet5Firs = () => {
+    setWaiter(true);
+
+    setTimeout(async () => {
+      setWaiter(false);
+
+      return setGoodsList(await get5First());
+    }, 500);
+  };
+
+  const handleGetRedGoods = () => {
+    setWaiter(true);
+
+    setTimeout(async () => {
+      setWaiter(false);
+
+      return setGoodsList(await getRedGoods());
+    }, 500);
+  };
```

Comment:
All these 3 functions do almost same job, create one handler function that will accept loader function, call it and set result to state